intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd

```diff
@@ -0,0 +1,507 @@
+  select(log_fold_change, t_statistic)
+```
+
+Let's say we wanted to arrange this dataset so that the genes are arranged by 
+the smallest p values to the largest.
+In order to do this, we would use the function `arrange()` as well as the column
+we would like to sort by (in this case `p_value`).
+
+```{r}
+stats_df %>% 
+  arrange(p_value) 
+```
+
+What if we want to sort from largest to smallest? 
+Like if we want to see the genes with the highest average expression?
+We can use the same function, but instead use the `desc()` function and now we 
+are using `avg_expression` column. 
+
+```{r}
+stats_df %>% 
+  arrange(desc(avg_expression))
+``` 
+
+What if we would like to create a new column of values?
+For that we use `mutate()` function.
+
+```{r}
+stats_df %>% 
+  mutate(log10_p_value = -log10(p_value))
+```
+
+What if we want to obtain summary statistics for a column? 
+The `summarize` function allows us to calculate summary statistics for a column. 
+Here we will use summarize to obtain an mean log folder change over all the 
+genes. 
+
+```{r}
+stats_df %>% 
+  summarize(mean(log_fold_change))
+```
+
+What if we'd like to obtain a summary statistics but have them for various 
+groups?
+Conveniently named, there's a function called `group_by()` that seamlessly 
+allows us to do this. 
+Also note that `group_by()` allows us to group by multiple variables at a time 
+if you want to.
+
+```{r}
+stats_summary_df <- stats_df %>%
+      group_by(contrast) %>% 
+      summarize(mean(log_fold_change))
+```
+
+Let's look at a preview of what we made:
+
+```{r}
+stats_summary_df
+```
+
+Here we have the mean log fold change expression per each contrast we made. 
+
+## A brief intro to the `apply` family of functions
+
+In base R there the `apply` family of functions can be an alternative 
+methods for performing transformations across a data frame, matrix or for other 
+object structures. 
+
+One of this family is (shockingly) the function `apply()`, which operates on 
+matrices or data frames. 
+The first argument to `apply()` is the data object we want to work on.
+The third argument is the function we will apply to each row or column of the 
+data object.
+The second argument in specifies whether we are applying the function 
+across rows or across columns (1 for rows, 2 for columns).
+
+Remember that `gene_df` is a gene x sample gene expression data frame. 
+
+```{r}
+# Calculate row means
+gene_means <- apply(gene_df[, -1], 1, mean) # Notice we are using 1 here
+
+# How long will `gene_means` be? 
+length(gene_means)
+```
+
+Why do we have a `[, -1]` after `gene_df` in the above chunk?
+
+Now let's investigate the same set up, but use 2 to `apply` over the columns of
+our matrix.
+
+```{r}
+# Calculate sample means
+sample_means <- apply(gene_df[, -1], 2, mean) # Notice we use 2 here
+
+# How long will `sample_means` be? 
+length(sample_means)
+```
+
+Although the `apply` functions may not be as easy to use as the `tidyverse` 
```

Comment:
I don't know whether we want to style "tidyverse" as `tidyverse`, but it seems like we were not doing that much, and we should try to be consistent. Perhaps we should refer to "the Tidyverse" in text, and `tidyverse` only when specifically refering to the package.
```suggestion
Although the `apply` functions may not be as easy to use as the tidyverse 
```